opm/simulators/linalg/extractMatrix.hpp

```diff
@@ -117,6 +117,32 @@ namespace Details
     }
 
 
+
+    template <class Matrix>
+    bool matrixEqual(const Matrix& m1, const Matrix& m2)
+    {
+        // Compare size and nonzeroes.
+        if (m1.N() != m1.N()) return false;
```

Comment:
Wow - that was hard to read :eyeglasses:

Reply:
Not to mention the fact that unless `m1.N()` is `NaN`, this condition will never be true.  I **suspect** one of the `m1`s should be an `m2`.

Reply:
As the commit message says: `untested`. With a test you would have discovered the `m1 != m1` issue which @bska pointed out (I did not see it). This is new code - not deeply entranched in the existing patterns - no reason to not make a test.

Reply:
And furthermore: Why not make a separate PR of these matrix utilities right away?

Reply:
Ouch! I might have made a test, and a PR, but this part is code thrown together in a debugging process, and not really intended for the end product. Still embarrassing to see...